cloud/amazon/ec2_group_facts.py

```diff
@@ -0,0 +1,272 @@
+            type: string
+        from_port:
+            description: The start of port range for the TCP and UDP protocols, or an ICMP type number. A value of -1 indicates all ICMP types.
+            type: int
+        to_port:
+            description: The end of port range for the TCP and UDP protocols, or an ICMP code. A value of -1 indicates all ICMP codes for the specified ICMP type.
+            type: int
+        user_id_group_pairs:
+            description: One or more security group and AWS account ID pairs.
+            type: list
+            contains:
+                user_id:
+                    description: ID of an AWS acccount
+                    type: string
+                group_name:
+                    description: The name of the security group.
+                    type: string
+                group_id:
+                    description: The ID of the security group.
+                    type: string
+                vpc_id:
+                    description: The ID of the VPC for the referenced security group, if applicable.
+                    type: string
+                vpc_peering_connection_id:
+                    description: The ID of the VPC peering connection, if applicable.
+                    type: string
+                peering_status:
+                    description: The status of a VPC peering connection, if applicable.
+                    type: string
+        ip_ranges:
+            description: One or more IP ranges.
+            type: list
+            contains:
+                cidr_ip:
+                    description: The CIDR range. You can either specify a CIDR range or a source security group, not both.
+                    type: string
+        prefix_list_ids:
+            description: One or more prefix list IDs for an AWS service.
+            type: list
+            contains:
+                prefix_list_id:
+                    description: The ID of the prefix.
+                    type: string
+owner_id:
+    description: AWS account ID of the security group owner
+    type: string
+    sample: 012345678901
+tags:
+    description: Any tags assigned to the security group
+    type: list
+    contains:
+        key:
+            description: The key of the tag.
+            type: string
+        value:
+            description: The value of the tag.
+            type: string
+vpc_id:
+    description: ID of the VPC for the security group
+    type: string
+    sample: vpc-12345678
+'''
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def list_ec2_groups(connection, module):
+
+    group_ids = module.params.get("group_ids")
+    group_names = module.params.get("group_names")
+    filters = ansible_dict_to_boto3_filter_list(module.params.get("filters"))
+
+    try:
+        all_sgs = connection.describe_security_groups(GroupIds=group_ids, GroupNames=group_names, Filters=filters)
```

Comment:
This conflicts with the documentation above. I recommend removing `GroupIds` and `GroupNames`.